test/junit/scala/tools/testing/AssertUtil.scala

```diff
@@ -97,4 +97,30 @@ object AssertUtil {
   def assert8(b: => Boolean, msg: => Any) =
     if (ScalaVersion(javaSpecVersion) == version8) assert(b, msg)
     else if (!b) println(s"assert not $msg")
+
+
+  /** Assert no new threads, with some margin for arbitrary threads to exit. */
+  def assertZeroNetThreads(body: => Unit): Unit = {
+    val beforeCount = Thread.activeCount
+    val beforeThreads = new Array[Thread](beforeCount)
+    assertEquals("Spurious early thread creation.", beforeCount, Thread.enumerate(beforeThreads))
```

Comment:
Might as well "wait for quiescence" here, too, as it doesn't hurt anything. But this should be done in a new ThreadGroup for isolation, since partest tries to run multiple tests in-process now, IIUC.

Reply:
Or just run this test in a forked jvm, we do it for some others. Just add a `.javaopts` file (https://github.com/scala/scala/blob/2.13.x/test/files/run/t2318.javaopts).

Reply:
I went with thread isolation.